### PR TITLE
Accordion

### PR DESCRIPTION
**Pull request by Aierie**
Demo of an accordion can be found on `/accordion`. This uses HTML from https://www.w3.org/TR/wai-aria-practices-1.1/examples/accordion/accordion.html as an accessible base to emulate motion of items in https://www.framer.com/examples/expandable-menu/. 

I opted to have the panels have individual animation contexts for this demo, since that seems simpler than putting the main accordion in charge of coordinating its panels, though for more complex data-driven choreography (one closing before another opens) that seems necessary?

### Some issues faced while creating this accordion
1. Modifying a class of a containing element (`local-class="Accordion {{if this.isFocused "focus"}}"`) seems to cancel animations that are in progress
2. I wasn't sure how to use the `hidden` HTML attribute like the original demo, for panels that are closed. It seems easy to use `{{#if}}` as a replacement though.
3. When the animation context's height changes based on its contents, it seems like appending a removed sprite to the orphans element and locking styles will set the removed sprite's original bounds to an unexpected position. I used a fixed height to do this - could also get the height of the header programmatically instead. However it feels like something that could be good for this library to handle? In the image below, the IP address, ZIP code, Last Meal and Pet's Name fields belong to the "Even more information" panel, that is about to close as the panel above it opens. Locking styles before or after appending the removed panel as an orphan will still result in an awkward starting position.
Snapshot of the accordion as the 'Even more information' panel closes. The contents of that panel were moved to an orphans element and are now sitting atop the header of that panel, which means that trying to calculate the panel's initial bounds will result in unexpected behaviour.

### CODE REVIEW

**Comment by nickschot** [comment]
1. Is an interesting point. We're almost certainly cancelling animations too eagerly in `willTransition`. The assumption was that for an interruption you'd always manually have to continue an animation, but an interruption/cancellation (from the low level implementation perspective) also happens when something completely unrelated like your if statement happens (such a change _may_ have an effect on a sprite of course but in this case doesn't).

**Comment by Aierie** [comment]
i went looking at how the cancelations actually happen. my understanding of what happens when an animation context detects a rerender is a bit clearer now, I think (though this is still somewhat of a simplification and is more focused on kept sprites): 

1. whenever there is a new rerender, we cancel all animations.
2. we calculate the new bounds of the sprites and animation contexts. 
3. we compare the new bounds against the previous bounds. the new bounds will become the previous bounds in the next rerender.
4. if there is a change, we start a new animation.
5. however if there is no change, we don't try to start a new animation. 

because the change calculation does not take into account the bounds of an element while being animated ("in-between" states are not recorded) but instead uses the final bounds, this means that a rerender in which the final bounds of a kept sprite remain the same will cancel an animation without starting a new one. 

~~in https://github.com/cardstack/animations-experiment/pull/24/commits/b533dafb29e264e4c3738474f942447e9fdcde7f i've moved the same logic used to check for changes in the transition runner to the animations service, and used it to decide whether or not to cancel an animation. this is here to demonstrate the fix, but i haven't fully gone through the side effects + perhaps it is good to organise stuff so that this change detection can be shared between the service and the runner? so i feel like it would be good in a separate PR.~~

This change is now removed.

**Comment by Aierie** [comment]
The video below demonstrates the cancelation problem using the interruptions demo. when i press the A key twice, the ball jumps to the A area abruptly, skipping the animation:

https://user-images.githubusercontent.com/39579264/159905222-e1280563-4299-47d2-8cf0-2bd4c449c79a.mov

**Comment by Aierie** [comment]
video of a bug that this change introduces - quickly pressing "A" then "D" while the ball is on "D" will mess up the sprite - it seems like this change breaks some assumptions that we're making about sprites always having initial bounds. 

https://user-images.githubusercontent.com/39579264/159908576-b22ac6a7-1c84-4070-ac7b-ca771c5ad489.mov


```
animations.js:156 Error: Assertion Failed: sprite should always have initialBounds
    at Object.assert (index.js:172:1)
    at TransitionRunner.maybeTransitionTask (transition-runner.js:115:1)
    at maybeTransitionTask.next (<anonymous>)
```

File: app/components/accordion/index.hbs [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+
+  The name and trademarks of copyright holders may NOT be used in advertising or publicity pertaining to the work without specific, written prior permission. Title to copyright in this work will at all times remain with copyright holders.
+
+  Notes
+  This version: http://www.w3.org/Consortium/Legal/2015/copyright-software-and-document
+
+  Previous version: http://www.w3.org/Consortium/Legal/2002/copyright-software-20021231
+
+  This version makes clear that the license is applicable to both software and text, by changing the name and substituting "work" for instances of "software and its documentation." It moves "notice of changes or modifications to the files" to the copyright notice, to make clear that the license is compatible with other liberal licenses.
+--}}
```
Comment (Aierie) [inline app/components/accordion/index.hbs:28]: @lukemelia is this license ok to include?

**Comment by nickschot** [comment]
If we merge #26 and rebase this PR it should fix the height issue we saw.

File: app/components/accordion/index.hbs (hunk 1 above)
Comment (lukemelia) [reply app/components/accordion/index.hbs:28]: That's fine.

**Pull request merged**